src/coreclr/src/vm/threads.cpp

```diff
@@ -54,6 +54,41 @@
 #include "eventpipebuffermanager.h"
 #endif // FEATURE_PERFTRACING
 
+static PortableTailCallFrame g_sentinelTailCallFrame = { NULL, NULL, NULL };
+
+TailCallTls::TailCallTls()
+    : m_frame(&g_sentinelTailCallFrame)
+    , m_argBuffer(NULL)
+    , m_argBufferSize(0)
+    , m_argBufferGCDesc(NULL)
+{
+}
+
+void* TailCallTls::AllocArgBuffer(size_t size, void* gcDesc)
+{
```

Comment:
It would be useful to these methods annotated with contracts. E.g. this one throws exception due to `new char[size]` call, but the caller did not account for it.

Reply:
Should I also annotate the rest of the functions in tailcallhelp.cpp?

Reply:
It would not hurt. The methods that are called during JITing should jus have `STANDARD_VM_CONTRACT` contract.